### ISSUE TITLE
feat: integrate CBE into DMRG sweep loop (1-site+CBE mode)

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#158]
## Feature Request

Wire `expand_bond` / `expand_bond_symmetric` into the DMRG sweep loop as a `subspace_expansion=True` option in `DMRGConfig`, enabling bond dimension growth in 1-site DMRG.

## Background

`expand_bond` and `expand_bond_symmetric` (PR #154, #157) implement the standalone CBE algorithm. The next step is integrating them into the DMRG sweep so that 1-site+CBE runs as a single `dmrg()` call.

## Current validation status

- Standalone CBE correctly produces expanded tensors with orthogonal new columns
- Block-sparse `expand_bond_symmetric` uses `_blockwise_contract` for fully block-sparse matvec
- Per-sector charge assignment works
- **Not yet validated**: full 1-site+CBE sweep converging to 2-site energy

## Integration challenges identified

1. **Boundary sites** (2-leg tensors): CBE expects 3-leg tensors. Need padding/unpadding.
2. **Label collision**: after SVD-truncating the expanded tensor and absorbing singular values into the next site, the bond labels can collide (label appears 3 times).
3. **Environment updates**: after CBE changes bond dimensions, left/right environments must be rebuilt with the new dimensions.
4. **Mixed canonical form**: CBE expansion breaks canonical form. Need QR to restore before environment update.

## Proposed API

```python
config = DMRGConfig(
    max_bond_dim=64,
    num_sweeps=20,
    two_site=False,           # 1-site mode
    subspace_expansion=True,  # enable CBE
    cbe_k=4,                  # expansion vectors per bond
)
result = dmrg(mpo, mps, config)
```

## Implementation sketch

In the sweep loop (`dmrg.py:215-250`), after `_one_site_update`:

```python
if config.subspace_expansion and not config.two_site:
    expanded = expand_bond_symmetric(site, l_env, W_l, W_r, r_env_2, next_site, k=config.cbe_k)
    A, s, B, _ = _svd_and_truncate_site(expanded, i, config)
    mps_tensors[i] = A
    mps_tensors[i+1] = B  # with absorbed singular values
```

This effectively turns 1-site+CBE into a "cheap 2-site" step: expand, then SVD-truncate back.

## References

- Standalone CBE: PR #154, #157
- Issue #152 (partial resolution — standalone function done, integration pending)
- McCulloch & Osborne, arXiv:2403.00562

**Comment by yingjerkao** [comment]
## Progress update

Branch `feat/cbe-integration` has the sweep integration framework:
- `DMRGConfig.subspace_expansion` and `cbe_k` fields added
- L→R sweep calls `_cbe_expand_site` + `_cbe_truncate_and_absorb`
- Boundary sites correctly skipped
- Dense SVD truncation + einsum absorption working

### Remaining blocker

After the dense SVD truncates the expanded tensor, the new bond needs charge labels. Currently uses `charges=zeros` (trivial), which creates wrong block structure for SymmetricTensor, giving E=0.

The fix: use `tenax.linalg.svd` (the Tensor-protocol SVD that derives per-sector charges automatically) instead of raw `jnp.linalg.svd`. But this requires the expanded tensor to already be a properly-constructed SymmetricTensor with correct charges on the expanded bond — which loops back to the charge assignment problem in `expand_bond_symmetric`.

### Possible approaches
1. Use `_truncated_svd_symmetric` directly on the expanded SymmetricTensor (if charges are correct)
2. Round-trip through dense for the entire CBE+truncate step (loses symmetry benefit but works)
3. Derive bond charges from the left/physical index charges using charge conservation

**Comment by yingjerkao** [comment]
## Key finding: truncation approach is wrong

The current CBE integration truncates the expanded site ALONE:
```
expanded(chi_l, d, chi_r+k) → SVD → A(chi_l, d, keep) + Vh(keep, chi_r+k)
```
Then `Vh[:, :chi_r] · neighbor` absorbs into the neighbor, discarding the expansion columns.

This is wrong: the SVD of the expanded site alone doesn't capture how the expansion vectors interact with the neighbor. The correct approach (matching 2-site DMRG):

```
theta = expanded(chi_l, d, chi_r+k) ⊗ neighbor(chi_r, d', chi_r')
      → pad neighbor to (chi_r+k, d', chi_r') 
      → SVD on theta → A(chi_l, d, keep) + B(keep, d', chi_r')
```

This requires either:
1. Padding the neighbor's left bond to chi_r+k (zeros for the expansion dims), forming the 2-site theta, then SVD
2. Or recognizing that the expansion dims have zero overlap with the neighbor and just using the QR of the expanded site to orthogonalize (without SVD truncation), then letting the NEXT sweep's 1-site optimization adjust

Option 2 is actually the standard CBE approach: don't truncate immediately, just expand and re-optimize. The truncation happens naturally when the bond dimension exceeds max_bond_dim after multiple expansions.

Branch `feat/cbe-integration` has the framework. The fix is to change `_cbe_truncate_and_absorb` to pad the neighbor and form a proper 2-site theta.
